### PR TITLE
Firearms navigation fixes

### PR DESCRIPTION
**Pull request by saruniitr**
## Change description

Fixes issues found during review,
 - When editing sporting shotgun either from Product summary or Product detail, we currently ask few additional questions instead of going directly to summary screen. Similarly for other fields
 - Clicking back on sporting shotgun and product type again causes endless loop
 - Changing product type from eg ammunition to firearms, it is possible to proceed with some questions having invalid value eg firearms replica - we are temporarily not allowing editing product type

### CODE REVIEW

File: exporter/templates/applications/goods/add-good-detail-summary.html [hunk 1]
```diff
@@ -21,9 +21,6 @@ <h1 class="govuk-heading-l">
 				{{ good.firearm_details.type.value|default_na }}
 			</dd>
 			<dd class="govuk-summary-list__actions">
```
Comment (saruniitr) [inline exporter/templates/applications/goods/add-good-detail-summary.html:23]: Removing the empty `dd` messing up the table format in the template

File: exporter/templates/goods/good.html [hunk 2]
```diff
@@ -296,13 +296,6 @@ <h3 class="govuk-heading-m">{% lcs 'goods.GoodPage.YOUR_GOOD' %}</h3>
 						{{ good.firearm_details.type.value|default_na }}
 					</dd>
 					<dd class="govuk-summary-list__actions">
```
Comment (saruniitr) [inline exporter/templates/goods/good.html:298]: same here

File: exporter/goods/services.py [hunk 3]
```diff
@@ -115,6 +115,19 @@ def add_firearm_details_to_data(json):
     if firearm_details and "is_sporting_shotgun" not in firearm_details:
         firearm_details["is_sporting_shotgun"] = False
 
+    if firearm_details and "year_of_manufacture" not in firearm_details:
```
Comment (saruniitr) [inline exporter/goods/services.py:118]: similar workaround as in `is_sporting_shotgun` above.

Comment (saruniitr) [inline exporter/goods/services.py:118]: similar workaround as in `is_sporting_shotgun` above.

File: exporter/goods/views.py [hunk 4]
```diff
@@ -234,23 +235,15 @@ def get_data(self):
...
-                return reverse_lazy("goods:good_military_use", kwargs={"pk": self.object_pk})
-        elif self.draft_pk:
+        if self.draft_pk:
             return reverse(
                 "goods:good_detail_application",
                 kwargs={"pk": self.object_pk, "type": "application", "draft_pk": self.draft_pk},
             )
-        # Edit
-        else:
+        elif self.application_id and self.object_pk:
```
Comment (saruniitr) [inline exporter/goods/views.py:243]: this is the case of coming from product summary screen and below `elif` is if we come from Product list -> product detail. Similar changes in all edit views of the fields.

File: exporter/goods/services.py (hunk 3 above)
Comment (r4vi) [reply exporter/goods/services.py:118]: it's valid not to send this as it's not a required field, why do we need to set to 0?

File: exporter/goods/services.py [hunk 5]
```diff
@@ -115,6 +115,19 @@ def add_firearm_details_to_data(json):
...
+        firearm_details["year_of_manufacture"] = 0
+
+    if firearm_details and "is_covered_by_firearm_act_section_one_two_or_five" not in firearm_details:
+        firearm_details["is_covered_by_firearm_act_section_one_two_or_five"] = False
```
Comment (r4vi) [inline exporter/goods/services.py:122]: not entirely comfortable with setting these values to defaults - is there any way to avoid this?

Comment (saruniitr) [reply exporter/goods/services.py]: added in the else case as discussed

**Review by r4vi — APPROVED** [review]

**Pull request merged**